lib/aws/AwsV4Signer.d.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { Credentials } from '@aws-sdk/types';
+import Connection from '../Connection';
+import * as http from 'http';
+
+interface AwsV4SignerOptions {
```

Comment:
So technically this is "AwsSigv4". I think we should align with other implementations such as `Aws::Sigv4::Signer` in Java and include "sig" and lowercase the "v", so `AwsSigv4Signer...`?

Reply:
Yeah, let's have them be consistent. I renamed everything.